File: web/playwright-csharp/Pages/BasePage.cs
using Microsoft.Playwright;

namespace Pages;

public class BasePage
{
    protected readonly IPage Page;

    public BasePage(IPage page)
    {
        Page = page;
    }

    public async Task NavigateToAsync(string url)
    {
        await Page.GotoAsync(url);
    }

    public async Task<ILocator> WaitForElementAsync(string selector)
    {
        return Page.Locator(selector);
    }
}

File: swag-labs/playwright-bdd-ts/src/pages/BasePage.ts
import type { Page } from '@playwright/test';

// Classe base para páginas, concentrando navegação e utilitários comuns.
export class BasePage {
  constructor(protected readonly page: Page) {}

  async navigate(path: string) {
    await this.page.goto(path, { waitUntil: 'domcontentloaded' });
  }

  async waitForVisible(selector: string, timeout = 10000) {
    await this.page.locator(selector).first().waitFor({ state: 'visible', timeout });
  }
}

